tests/jobs/ahm_analysis/test_ahm_analysis.py

```diff
@@ -0,0 +1,418 @@
+import os
+
+import pytest
+import pandas as pd
+import numpy as np
+
+import semeio.workflows.ahm_analysis.ahmanalysis as ahmanalysis
+from semeio._exceptions.exceptions import ValidationError
+
+from unittest.mock import MagicMock
+from itertools import product
+from pathlib import Path
+
+from scipy import stats
+from ecl.grid import EclGridGenerator
+
+
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_make_update_log_df(test_data_root):
+    """test function creates a dataframe from update_log file
+    and replace '...' with key_obs"""
+    test_data_dir = os.path.join(test_data_root, "update_log/oneobs")
+    update_log_path = test_data_dir
+    key_obs = "RWI_3_OBS"
+    updatelog_obs = ahmanalysis.make_update_log_df(update_log_path, key_obs)
+    assert "..." not in updatelog_obs["obs_key"]
+    assert updatelog_obs.at[0, "status"] == "Active"
+    assert updatelog_obs.at[1, "status"] == "Inactive"
+    assert (
+        updatelog_obs.columns
+        == [
+            "obs_number",
+            "obs_key",
+            "obs_mean",
+            "obs_std",
+            "status",
+            "sim_mean",
+            "sim_std",
+        ]
+    ).all()
+
+
+@pytest.mark.parametrize(
+    "input_dir, obs_key, expected_result",
+    [
+        ("update_log/oneobs", "RWI_3_OBS", "2 active/3"),
+        ("update_log/allobs", "ALL_OBS", "41 active/47"),
+        ("update_log/noactive", "RWI_2_OBS", "0 active/3"),
+    ],
+)
+@pytest.mark.usefixtures("setup_tmpdir")
+def test_list_active_observations(test_data_root, input_dir, expected_result, obs_key):
+    """test function creates a dataframe reporting active observations during update"""
+    test_data_dir = os.path.join(test_data_root, input_dir)
+    df_update_log = ahmanalysis.make_update_log_df(test_data_dir, obs_key)
+    result = ahmanalysis.list_active_observations(df_update_log)
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "input_obs, expected_misfit, update_log",
+    [
+        (["RWI_3_OBS"], 2.0, {"status": {0: "Active", 1: "Inactive", 2: "Active"}}),
+        (
+            ["RWI_3_OBS", "OP_3_WWCT1"],
+            7.0 / 4.0,
+            {"status": {0: "Active", 1: "Inactive", 2: "Active", 3: "Active"}},
+        ),
+        (["OP_3_WWCT1"], 1.0, {"status": {0: "Active"}}),
+    ],
+)
+def test_list_observations_misfit(input_obs, expected_misfit, update_log):
+    """test function creates a dataframe
+    reporting misfit data for each obs vector except the ones with All_obs-"""
```

Comment:
There is nothing in the test, nor the function itself that say anything about `All_obs-`? But there shouldn't be either, right?

Reply:
the part except the ones with All_obs has been removed (should not be there)